test/end2end_test.go

```diff
@@ -193,6 +193,16 @@ func (s *testServer) HalfDuplexCall(stream testpb.TestService_HalfDuplexCallServ
 
 const tlsDir = "testdata/"
 
+func TestDialTimeout(t *testing.T) {
+	conn, err := grpc.Dial("localhost:0", grpc.WithTimeout(5*time.Microsecond))
```

Comment:
I guess this test is probably fine because the error I just described should be classified as Temporary so that grpc will keep reconnecting.